### PR TITLE
add a HTTP/2 client example

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

Making an HTTP/2 request is not easy, see also: https://stackoverflow.com/questions/53466997/using-swiftnio-and-swiftniohttp2-as-an-http2-client

Modifications:

Add a very simple HTTP/2 client.

Result:

Users know how to establish an HTTP/2 connection as a client.

### CODE REVIEW

**Comment by weissi** [comment]
CC'ing @tanner0101 and @MrMage as they might be interested in an example

**Comment by MrMage** [comment]
Thank you! I have briefly looked into spinning up an HTTP2 client and this looks roughly like what I expected.

One question: It looks like you are attaching a dedicated `CreateRequestStreamHandler` whose whole purpose is to spin open a multiplexed subchannel and send the request. As we are planning to send multiple requests via the same connection, is it possible to do that from outside a dedicated `CreateRequestStreamHandler`, e.g. by simply calling `multiplexer.createStreamChannel(promise: nil, requestStreamInitializer)` directly? Should we be observing the promise passed into that method in a production environment, or would we get notified about failures elsewhere as well?

While at it — in case the HTTP2 connection gets interrupted somehow, how should we go best about re-establishing a new one for subsequent requests? Simply spin up a new one via a fresh `ClientBootstrap` and use the new one from then on?

File: http2-client/Package.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import PackageDescription
+
+let package = Package(
+    name: "http2-client",
+    dependencies: [
+        // Dependencies declare other packages that this package depends on.
+        .package(url: "https://github.com/apple/swift-nio", from: "1.0.0"),
+        .package(url: "https://github.com/apple/swift-nio-ssl", from: "1.0.0"),
+        .package(url: "https://github.com/apple/swift-nio-http2", from: "0.1.0"),
```
Comment (Lukasa) [inline http2-client/Package.swift]: Using a `from` dependency on a 0. release is kinda asking for trouble ;)

Comment (weissi) [reply http2-client/Package.swift]: @Lukasa thanks, fixed

Comment (weissi) [reply http2-client/Package.swift]: @Lukasa you mention 1 comment in the diff, was that this one?

Comment (Lukasa) [reply http2-client/Package.swift]: Sure was, go ahead and merge.

**Pull request merged**